### DISCUSSION TITLE
linking the author location in profile/meta.html

### DISCUSSION THREAD

**Discussion by chrisrummel** [razonyang/hugo-theme-bootstrap#266] [Ideas]
I've tweaked layouts/partials/sidebar/profile/meta.html to turn the authors location, as given in config/_default/author.toml, into a link to a map-service. I've hardcoded OpenStreetMap, but it could easily be changed to something else, like GMaps, and it could even be configured in the config.

Would that be interesting so I should push it?

**Comment by razonyang** [comment]
https://github.com/razonyang/hugo-theme-bootstrap/blob/d8b5f2d5c5c47008624d841453cf31cf20616c32/layouts/partials/sidebar/profile/meta.html#L10

I think to modify the above code as following will solve this issue:

```html
<div class="profile-location"><i class="fas fa-fw fa-map-marker-alt"></i>{{ . | safeHTML }}</div>
```

It treats the location as safe HTML, so that the location can contain any valid HTML tags. Does this met your need?

PR are welcome.

**Reply by chrisrummel** [reply]
Thanks for the safeHTML-hint, which surely is necessary with locations containing special chars or spaces.

But my idea was turning the location from simply being a string into e.g.

`<div class="profile-location"><i class="fas fa-fw fa-map-marker-alt"></i><a href="https://www.openstreetmap.org/search?query={{ . | safeHTML }}">{{ . | safeHTML }}</a></div>`

**Reply by chrisrummel** [reply]
Oh, now I see what you meant:

`location = "<a href='https://www.openstreetmap.org/search?query=Earth'>Earth</a>"`

Well, of course that's one solution. Might be more userfriendly to add a couple conditions to only output location if no locationService is set, but if one is set, adding a link to the profile. But this works, too.

**Reply by razonyang** [reply]
Yep, that is what I mean.

IMO, the solution you provided is too complex. As for a user, I need to specify both of the locationService and location. I want to keep it as simple as possible, so that user just need to copy and paste.
